### PR TITLE
python: add LCS command

### PR DESCRIPTION
**Pull request by jamesx-improving**
Please comment

### CODE REVIEW

File: python/python/glide/async_commands/core.py [hunk 1]
```diff
@@ -5428,3 +5428,152 @@ def try_get_pubsub_message(self) -> Optional[PubSubMsg]:
...
+            key2 (str): The key that stores the second string.
+
+        Returns:
+            The length of the longest common subsequence between the 2 strings.
+
+        Examples:
+            >>> await client.mset({"testKey1" : "abcd", "testKey2": "axcd"})
+                'OK'
+            >>> await client.lcs_len("testKey1", "testKey2")
+                3  # the length of the longest common subsequence between these 2 stings is 2.
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:5484]:
```suggestion
                3  # the length of the longest common subsequence between these 2 strings is 2.
```

File: python/python/glide/async_commands/core.py [hunk 2]
```diff
@@ -5428,3 +5428,152 @@ def try_get_pubsub_message(self) -> Optional[PubSubMsg]:
...
+        See https://valkey.io/commands/lcs for more details.
+
+        Args:
+            key1 (str): The key that stores the first string.
+            key2 (str): The key that stores the second string.
+            min_match_len (Optional[int]): The minimum length of matches to include in the result.
+            with_match_len (Optional[bool]): If True, include the length of the substring matched for each substring.
+
+        Returns:
+            A Map containing the indices of the longest common subsequence between the
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:5513]: mapping or dict

File: python/python/glide/async_commands/core.py [hunk 3]
```diff
@@ -5428,3 +5428,152 @@ def try_get_pubsub_message(self) -> Optional[PubSubMsg]:
...
+                    ],
+                    'len': 7
+                }
+            >>> await client.lcs_idx("testKey1", "testKey2", with_match_len=True)
+                {
+                    'matches': [
+                        [
+                            [4, 7],
+                            [5, 8],
+                            4,  # length of this match ("1234")
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:5555]: inline comments in a sample output are really cool

File: python/python/tests/test_transaction.py [hunk 4]
```diff
@@ -604,6 +606,21 @@ async def transaction_test(
         transaction.bzmpop([key16], ScoreFilter.MIN, 0.1, 2)
         args.append([key16, {"a": 1.0, "b": 2.0}])
 
+        transaction.set(key23, "abcd1234")
+        args.append(OK)
+        transaction.set(key24, "bcdef1234")
+        args.append(OK)
```
Comment (Yury-Fridlyand) [inline python/python/tests/test_transaction.py:612]: You can use mset as you show in example

**Review by Yury-Fridlyand — APPROVED** [review]

File: python/python/glide/async_commands/core.py (hunk 3 above)
Comment (jamesx-improving) [reply python/python/glide/async_commands/core.py:5555]: 😊

File: python/python/glide/async_commands/core.py [hunk 5]
```diff
@@ -5428,3 +5428,152 @@ def try_get_pubsub_message(self) -> Optional[PubSubMsg]:
...
+            key2 (str): The key that stores the second string.
+
+        Returns:
+            The length of the longest common subsequence between the 2 strings.
+
+        Examples:
+            >>> await client.mset({"testKey1" : "abcd", "testKey2": "axcd"})
+                'OK'
+            >>> await client.lcs_len("testKey1", "testKey2")
+                3  # the length of the longest common subsequence between these 2 strings is 2.
```
Comment (GumpacG) [inline python/python/glide/async_commands/core.py:5484]: Should it be?
```suggestion
                3  # the length of the longest common subsequence between these 2 strings is 3.
```

File: python/python/tests/test_async_client.py [hunk 6]
```diff
@@ -7102,6 +7102,119 @@ async def test_standalone_client_random_key(self, redis_client: GlideClient):
...
+        assert await redis_client.set(key2, value2) == OK
+        assert await redis_client.lcs_len(key1, key2) == expected_subsequence_length
+        assert (
+            await redis_client.lcs_len(key1, nonexistent_key)
+            == expected_subsequence_length_with_nonexistent_key
+        )
+
+    @pytest.mark.parametrize("cluster_mode", [False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_lcs_len_idx(self, redis_client: GlideClient):
```
Comment (GumpacG) [inline python/python/tests/test_async_client.py:7149]:
```suggestion
    async def test_lcs_idx(self, redis_client: GlideClient):
```

File: python/python/tests/test_async_client.py [hunk 7]
```diff
@@ -7102,6 +7102,119 @@ async def test_standalone_client_random_key(self, redis_client: GlideClient):
...
+                    3,
+                ],
+            ],
+            "len": 7,
+        }
+        expected_response_with_nonexistent_key = {
+            "matches": [],
+            "len": 0,
+        }
+        assert await redis_client.set(key1, value1) == OK
```
Comment (GumpacG) [inline python/python/tests/test_async_client.py:7199]: I think we need test cases for:
- LCS with IDX and a negative MINMATCHLEN
- LCS with IDX, MINMATCHLEN, and WITHMATCHLEN
- non-string keys are used

File: python/python/glide/async_commands/core.py (hunk 5 above)
Comment (jduo) [reply python/python/glide/async_commands/core.py:5484]: The longest sequence between 'abcd' and 'axcd' should be 2 (the 'cd' sequence), so we should return 2:
```suggestion
                2  # the length of the longest common subsequence between these 2 strings is 2 (the "cd" sequence).
```

File: python/python/glide/async_commands/transaction.py [hunk 8]
```diff
@@ -3891,6 +3891,89 @@ def random_key(self: TTransaction) -> TTransaction:
...
+
+        Args:
+            key1 (str): The key that stores the first string.
+            key2 (str): The key that stores the second string.
+
+        Command Response:
+            A String containing the longest common subsequence between the 2 strings.
+            An empty String is returned if the keys do not exist or have no common subsequences.
+
+        Since: Redis version 7.0.0.
```
Comment (jduo) [inline python/python/glide/async_commands/transaction.py:3911]: This needs a Returns description and an example.

File: python/python/tests/test_async_client.py [hunk 9]
```diff
@@ -7102,6 +7102,119 @@ async def test_standalone_client_random_key(self, redis_client: GlideClient):
...
+    async def test_lcs(self, redis_client: GlideClient):
+        min_version = "7.0.0"
+        if await check_if_server_version_lt(redis_client, min_version):
+            return pytest.mark.skip(reason=f"Redis version required >= {min_version}")
+        key1 = "testKey1"
+        value1 = "abcd"
+        key2 = "testKey2"
+        value2 = "axcd"
+        nonexistent_key = "nonexistent_key"
+        expected_subsequence = "acd"
```
Comment (jduo) [inline python/python/tests/test_async_client.py:7116]: I don't get this -- why is acd the longest common subsequence? Shouldn't it be cd?

Comment (jamesx-improving) [reply python/python/tests/test_async_client.py:7116]: This is `Longest common subsequence`, not to be confused with `Longest common substring`.

File: python/python/glide/async_commands/core.py (hunk 5 above)
Comment (jamesx-improving) [reply python/python/glide/async_commands/core.py]: Like I mentioned in the other reply, `Longest common subsequence` and `Longest common substring` are defined differently.

**Review by GumpacG — APPROVED** [review]

**Pull request merged**